ambry-api/src/main/java/com.github.ambry/account/Account.java

```diff
@@ -0,0 +1,312 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * <p>
+ *   A representation of an Ambry user. A user is an entity (an application or an individual user) who uses ambry
+ * as a service. This {@code Account} class contains general information of a user, which can be used for user-based
+ * operations such as authentication, get a {@link Container} under this account, and access control.
+ * </p>
+ * <p>
+ *   Account name is provided by an Ambry user as an external reference. Account id is an internal identifier
+ *   of the user, and is one-to-one mapped to an account name. Account name and id are generated through user
+ *   registration process.
+ * </p>
+ * <p>
+ *   Account is serialized into {@link JSONObject} in the highest metadata version, which is version 1 for now.
+ *   Below lists all the metadata versions and their formats:
+ * </p>
+ * <pre><code>
+ * Version 1:
+ * {
+ *   "accountId": 101,
+ *   "accountName": "MyAccount",
+ *   "containers": [
+ *     {
+ *       "containerName": "MyPrivateContainer",
+ *       "description": "This is my private container",
+ *       "isPrivate": "true",
+ *       "containerId": 0,
+ *       "version": 1,
+ *       "status": "ACTIVE",
+ *       "parentAccountId": "101"
+ *     },
+ *     {
+ *       "containerName": "MyPublicContainer",
+ *       "description": "This is my public container",
+ *       "isPrivate": "false",
+ *       "containerId": 1,
+ *       "version": 1,
+ *       "status": "ACTIVE",
+ *       "parentAccountId": "101"
+ *     }
+ *   ],
+ *   "version": 1,
+ *   "status": "ACTIVE"
+ * }
+ * </code></pre>
+ * <p>
+ *   An account object is immutable. To update an account, refer to {@link AccountBuilder} for how to build a new
+ *   account object with updated field(s).
+ * </p>
+ */
+public class Account {
+  // static variables
+  static final String ACCOUNT_METADATA_VERSION_KEY = "version";
+  static final String ACCOUNT_ID_KEY = "accountId";
+  static final String ACCOUNT_NAME_KEY = "accountName";
+  static final String ACCOUNT_STATUS_KEY = "status";
+  static final String CONTAINERS_KEY = "containers";
+  static final short ACCOUNT_METADATA_VERSION_1 = 1;
+  static final short HIGHEST_ACCOUNT_METADATA_VERSION = ACCOUNT_METADATA_VERSION_1;
```

Comment:
can you name this CURRENT_VERSION. Also for other variables, why do we need "metadata" prefixes. Just "version" should suffice since anyone would call it as "Account.Version" or "Account.Current_Version" sort of